### PR TITLE
User deactivation feature

### PR DESCRIPTION
**Pull request by lynguyxn**
### Administrative Info
Monday Board ID: 1181559666
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added confirm deactivation popup on frontend when user clicks "Deactivate"
- Created backend delete user route and linked it to the frontend for the deactivate button
- Redirect user to Roster page after deactivating another account or Login page after deactivating their own account
- Users with either roster edit or account activation permissions can see settings icon
- Users with roster edit permissions (currently "admin" permissions before Justin's PR) but not deactivate permissions can only see "Edit" button

### Testing
How did you confirm your changes worked? 
- Logged in as president, deleted developer account, and checked if the developer account is deleted from database
- Logged in as user with edit but no delete permission and checked it only showed the "Edit" button

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Deactivation Pop-up |
| --- |
|  |

| President View - Redirect to Roster after deleting Developer account |
| --- |
|  |

| Database deletion of Developer account |
| --- |
|  |

| No deactivate but has roster edit permissions view |
| --- |
| image |

| Redirect to Login after deleting their own account |
| --- |
| image |

### CODE REVIEW

**Review by justinyaodu — CHANGES_REQUESTED** [review]
When a user has edit permissions but not deactivate permissions, can we show only the edit option in the menu? I feel like that might be less confusing than the popup. Other than that, LGTM!

**Review by hahsu — COMMENTED** [review]
I'll look at this after Justin's PR is merged into main

**Comment by lynguyxn** [comment]
> When a user has edit permissions but not deactivate permissions, can we show only the edit option in the menu? I feel like that might be less confusing than the popup. Other than that, LGTM!

Good idea, made the changes :)

**Review by justinyaodu — APPROVED** [review]
LGTM!

File: backend/services/users.js [hunk 1]
```diff
@@ -121,11 +121,16 @@ async function editUser(rawUser, editingUser) {
   return editedUser.save();
 }
 
+async function deleteUser(_id) {
```
Comment (hahsu) [inline backend/services/users.js:124]: Can you add a header comment like the rest of the functions have in this file?

**Review by hahsu — APPROVED** [review]
lgtm other than the one comment!

**Pull request merged**